app/controllers/JSONDataFile.scala

```diff
@@ -0,0 +1,37 @@
+package controllers
+
+import akka.stream.Materializer
+import akka.stream.scaladsl.FileIO
+import play.api.Configuration
+import play.api.http.HttpEntity
+import play.api.libs.circe.Circe
+import play.api.mvc.{AbstractController, ControllerComponents, ResponseHeader, Result}
+import auth.{BearerTokenAuth, Security}
+import play.api.cache.SyncCacheApi
+import java.nio.file.{Files, Paths}
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+
+
+class JSONDataFile @Inject() (cc:ControllerComponents, override implicit val config:Configuration, override val bearerTokenAuth:BearerTokenAuth)(implicit mat:Materializer, ec:ExecutionContext, override implicit val cache:SyncCacheApi) extends AbstractController(cc) with Security with Circe {
+  def getAFile(filePath:String)= IsAuthenticated { uid =>
+    request => {
+      val fullPath = Paths.get(config.get[String]("vaults.vaultDataPath"), filePath + ".json")
```

Comment:
You should sanitise the filepath first (I suggest filtering out all `/` since we don't need them) - as things stand you could put in say `../../etc/passwd` for example and extract confidential information

Reply:
I do not think that is possible because the code forces '.json' on to the end of the path.

Reply:
Done.